lab/skel_graph/traversal/_examples/scc/kosaraju/main.cpp

```diff
@@ -1,61 +1,93 @@
-//Componente tare conexe - Kosaraju - O(N+M)
+// Copyright Popescu Alexandru Gabriel <alexandru.popescu0697@gmail.com> 2018
+// Copyright Darius Neatu <neatudarius@gmail.com> 2018
+// Componente tare conexe - Kosaraju - O(n + m)
 // https://infoarena.ro/problema/ctc
+#include <bits/stdc++.h>
 
-#include <fstream>
-#include <bitset>
-#include <vector>
-#define Nmax 100099
+#define NMAX 100001
 #define pb push_back
-using namespace std;
-ifstream f("ctc.in");
-ofstream g("ctc.out");
- 
-int N,M,sol,Tsort[Nmax],x,y;
-bitset < Nmax> viz;
-vector < int > G[Nmax],T[Nmax],CTC[Nmax];
- 
- 
-void PrintOutput()
-{
- 
-}
- 
-void DFS(int node)
-{
-     viz[node]=1;
-     for(vector<int>::iterator it=G[node].begin();it!=G[node].end();++it)
-          if(!viz[*it])DFS(*it);
-     Tsort[++Tsort[0]]=node;
-}
- 
-void DFS_T(int node)
-{
-     viz[node]=0; CTC[sol].pb(node);
-     for(vector<int>::iterator it=T[node].begin();it!=T[node].end();++it)
-          if(viz[*it])DFS_T(*it);
-}
- 
-void Tarjan()
-{
-     for(int i=1; i<=N; ++i)
-          if(!viz[i])DFS(i);
-     for(int i=N; i>=1; --i)
-          if(viz[Tsort[i]])
-                ++sol , DFS_T(Tsort[i]);
+std::ifstream f("ctc.in");
+
+class Task {
+public:
+    void solve() {
+        read_input();
+        kosaraju();
+        print_output();
+    }
+
+private:
+    int n, m, sol = 0;
```

Comment:
Let's keep this convention:

// n = numar de noduri, m = numar de muchii
int n, m;

// adj[node] = lista de adicenta a lui node
vector<int> adj[NMAX];

int sol = 0;

(other stuff)